src/nvector/parallel/nvector_parallel.c

```diff
@@ -1420,31 +1430,29 @@ int N_VWrmsNormMaskVectorArray_Parallel(int nvec, N_Vector* X, N_Vector* W,
         nrm[i] += SUNSQR(xd[j] * wd[j]);
     }
   }
-  retval = MPI_Allreduce(MPI_IN_PLACE, nrm, nvec, MPI_SUNREALTYPE, MPI_SUM, comm);
+  SUNCheckMPICallNoRet(MPI_Allreduce(MPI_IN_PLACE, nrm, nvec, MPI_SUNREALTYPE, MPI_SUM, comm));
 
   for (i=0; i<nvec; i++)
     nrm[i] = SUNRsqrt(nrm[i]/Ng);
 
-  return retval == MPI_SUCCESS ? 0 : -1;
+  return SUN_SUCCESS;
```

Comment:
I find it interesting that these changes eliminate the possibility of catching a non-`MPI_SUCCESS` return value from `MPI_Allreduce` when SUNDIALS is configured to disable error checking.  This same type of change occurs with other routines in this file.

Perhaps this is the point -- when turning off that CMake option we want things to run as fast as possible?  However, if that is the case then I'll note that we have many other non-recoverable failure flags that are not `#ifdef`ed out based on this CMake option.  

It would be nice if a future effort gave more nuanced control over the level of error checking (everywhere, where it existed before, none).